LLama/Native/Load/NativeLibraryConfig.cs

```diff
@@ -0,0 +1,605 @@
+            var vbmi = System.Runtime.Intrinsics.X86.Avx512Vbmi.IsSupported;
+#else
+            var f = (ebx & (1 << 16)) != 0;
+            var bw = (ebx & (1 << 30)) != 0;
+            var vbmi = (ecx & 0b_0000_0000_0010) != 0;
+#endif
+
+            return vnni && vbmi && bw && f;
+        }
+
+        /// <summary>
+        /// The description of the native library configurations that's already specified.
+        /// </summary>
+        /// <param name="Path"></param>
+        /// <param name="Library"></param>
+        /// <param name="UseCuda"></param>
+        /// <param name="AvxLevel"></param>
+        /// <param name="AllowFallback"></param>
+        /// <param name="SkipCheck"></param>
+        /// <param name="SearchDirectories"></param>
+        public record Description(string? Path, NativeLibraryName Library, bool UseCuda, AvxLevel AvxLevel, bool AllowFallback, bool SkipCheck, 
+            string[] SearchDirectories)
+        {
+            /// <inheritdoc/>
+            public override string ToString()
+            {
+                string avxLevelString = AvxLevel switch
+                {
+                    AvxLevel.None => "NoAVX",
+                    AvxLevel.Avx => "AVX",
+                    AvxLevel.Avx2 => "AVX2",
+                    AvxLevel.Avx512 => "AVX512",
+                    _ => "Unknown"
+                };
+
+                string searchDirectoriesString = "{ " + string.Join(", ", SearchDirectories) + " }";
+
+                return $"NativeLibraryConfig Description:\n" +
+                       $"- LibraryName: {Library}\n" +
+                       $"- Path: '{Path}'\n" +
+                       $"- PreferCuda: {UseCuda}\n" +
+                       $"- PreferredAvxLevel: {avxLevelString}\n" +
+                       $"- AllowFallback: {AllowFallback}\n" +
+                       $"- SkipCheck: {SkipCheck}\n" +
+                       $"- SearchDirectories and Priorities: {searchDirectoriesString}";
+            }
+        }
+    }
+#endif
+
+    public sealed partial class NativeLibraryConfig
+    {
+        /// <summary>
+        /// Set configurations for all the native libraries, including LLama and LLava
+        /// </summary>
+        [Obsolete("Please use NativeLibraryConfig.All instead, or set configurations for NativeLibraryConfig.LLama and NativeLibraryConfig.LLavaShared respectively.")]
+        public static NativeLibraryConfigContainer Instance { get; }
+
+        /// <summary>
+        /// Set configurations for all the native libraries, including LLama and LLava
+        /// </summary>
+        public static NativeLibraryConfigContainer All { get; }
+
+        /// <summary>
+        /// Configuration for LLama native library
+        /// </summary>
+        public static NativeLibraryConfig LLama { get; }
+
+        /// <summary>
+        /// Configuration for LLava native library
+        /// </summary>
+        public static NativeLibraryConfig LLavaShared { get; }
```

Comment:
Should we just call it `LLava` instead?

Reply:
It's okay to me. @SignalRT Any idea about this since you added the support for LLaVA?

Reply:
It's ok to me. The reason was the name of the library used in llama.cpp.